openapi-cli/src/main/java/io/ballerina/openapi/cmd/Flatten.java

```diff
@@ -0,0 +1,359 @@
+        String outputFileNameWithExt = getOutputFileName();
+        try {
+            CodegenUtils.writeFile(targetPath.resolve(outputFileNameWithExt),
+                    outputFileNameWithExt.endsWith(JSON_EXTENSION) ? Json.pretty(openAPI) : Yaml.pretty(openAPI));
+            infoStream.printf(INFO_OUTPUT_WRITTEN_MSG, targetPath.resolve(outputFileNameWithExt));
+        } catch (IOException exception) {
+            errorStream.println(ERROR_OCCURRED_WHILE_WRITING_THE_OUTPUT_OPENAPI_FILE);
+            exitError();
+        }
+    }
+
+    private String getOutputFileName() {
+        return resolveContractFileName(targetPath, fileName + getFileExtension(), format.equals("json"));
+    }
+
+    private String getFileExtension() {
+        return (Objects.nonNull(format) && format.equals("json")) ? JSON_EXTENSION : YAML_EXTENSION;
+    }
+
+    private void filterOpenAPIOperations(OpenAPI openAPI) {
+        Filter filter = getFilter();
+        if (filter.getOperations().isEmpty() && filter.getTags().isEmpty()) {
+            return;
+        }
+
+        // Remove the operations which are not present in the filter
+        openAPI.getPaths().forEach((path, pathItem) -> pathItem.readOperationsMap()
+                .forEach((httpMethod, operation) -> {
+                    if (!filter.getOperations().contains(operation.getOperationId()) &&
+                            operation.getTags().stream().noneMatch(filter.getTags()::contains)) {
+                        pathItem.operation(httpMethod, null);
+                    }
+                })
+        );
+
+        // Remove the paths which do not have any operations after filtering
+        List<String> pathsToRemove = new ArrayList<>();
+        openAPI.getPaths().forEach((path, pathItem) -> {
+            if (pathItem.readOperationsMap().isEmpty()) {
+                pathsToRemove.add(path);
+            }
+        });
+        pathsToRemove.forEach(openAPI.getPaths()::remove);
+    }
+
+    private Optional<OpenAPI> sanitizeOpenAPI(OpenAPI openAPI, List<String> existingComponentNames) {
+        Map<String, String> proposedNameMapping = getProposedNameMapping(openAPI, existingComponentNames);
+        if (proposedNameMapping.isEmpty()) {
+            return Optional.of(openAPI);
+        }
+
+        SwaggerParseResult parserResult = OASModifier.getOASWithSchemaNameModification(openAPI, proposedNameMapping);
+        openAPI = parserResult.getOpenAPI();
+        if (Objects.isNull(openAPI)) {
+            errorStream.println(ERROR_OCCURRED_WHILE_GENERATING_SCHEMA_NAMES);
+            if (!parserResult.getMessages().isEmpty()) {
+                errorStream.println(FOUND_PARSER_DIAGNOSTICS);
+                parserResult.getMessages().forEach(errorStream::println);
+            }
+            return Optional.empty();
+        }
+
+        return Optional.of(openAPI);
+    }
+
+    public Map<String, String> getProposedNameMapping(OpenAPI openapi, List<String> existingComponentNames) {
```

Comment:
why can't we use the same `getProposedNameMapping` function in OASModifer for this?

Reply:
Here, we are sanitizing the names generated with the flatten options, so there is additional logic to filter the names to get a proposed map